packages/core/src/QuestionGenerator.ts

```diff
@@ -7,22 +7,42 @@ import {
 import { BaseQuestionGenerator, SubQuestion } from "./engines/query/types";
 import { OpenAI } from "./llm/LLM";
 import { LLM } from "./llm/types";
+import { PromptMixin } from "./prompts";
 import { BaseOutputParser, StructuredOutput, ToolMetadata } from "./types";
 
 /**
  * LLMQuestionGenerator uses the LLM to generate new questions for the LLM using tools and a user query.
  */
-export class LLMQuestionGenerator implements BaseQuestionGenerator {
+export class LLMQuestionGenerator
+  extends PromptMixin
+  implements BaseQuestionGenerator
+{
   llm: LLM;
   prompt: SubQuestionPrompt;
   outputParser: BaseOutputParser<StructuredOutput<SubQuestion[]>>;
 
   constructor(init?: Partial<LLMQuestionGenerator>) {
+    super();
+
     this.llm = init?.llm ?? new OpenAI();
     this.prompt = init?.prompt ?? defaultSubQuestionPrompt;
     this.outputParser = init?.outputParser ?? new SubQuestionOutputParser();
   }
 
+  protected _getPrompts(): { [x: string]: SubQuestionPrompt } {
+    return {
+      subQuestion: this.prompt,
+    };
+  }
+
+  protected _updatePrompts(promptsDict: {
+    [x: string]: SubQuestionPrompt;
```

Comment:
Since you're looking for a specific prompt you can set the type variable name explicitly.